src/backend/distributed/utils/replication_origin_session_utils.c

```diff
@@ -0,0 +1,232 @@
+}
+
+
+/*
+ * SetupReplicationOriginRemoteSession sets up a new replication origin session in a
+ * remote session. The identifier is used to create a unique replication origin name
+ * for the session in the remote node.
+ */
+void
+SetupReplicationOriginRemoteSession(MultiConnection *connection)
+{
+	/*elog(LOG, "Setting up remote replication origin session %p %s,%d", connection, connection->hostname, connection->port); */
+	if (connection != NULL && !IsReplicationOriginSessionSetup(connection))
+	{
+		/*elog(LOG, "After IsReplicationOriginSessionSetup session %s,%d", connection->hostname, connection->port); */
+		StringInfo replicationOriginSessionSetupQuery = makeStringInfo();
+		appendStringInfo(replicationOriginSessionSetupQuery,
+						 "select pg_catalog.replication_origin_session_start_no_publish();");
+		ExecuteCriticalRemoteCommand(connection,
+									 replicationOriginSessionSetupQuery->data);
+		connection->isReplicationOriginSessionSetup = true;
+
+		/*elog(LOG, "After Setting up remote replication origin session %s,%d", connection->hostname, connection->port); */
+	}
+}
+
+
+/*
+ * ResetReplicationOriginRemoteSession resets the replication origin session in a
+ * remote node.
+ */
+void
+ResetReplicationOriginRemoteSession(MultiConnection *connection)
+{
+	if (connection != NULL && connection->isReplicationOriginSessionSetup)
+	{
+		/*elog(LOG, "Resetting remote replication origin session %s,%d", connection->hostname, connection->port); */
+
+		/* Reset the replication origin in remote session*/
+		StringInfo replicationOriginSessionResetQuery = makeStringInfo();
+		appendStringInfo(replicationOriginSessionResetQuery,
+						 "select pg_catalog.replication_origin_session_end_no_publish();");
+		ExecuteCriticalRemoteCommand(connection,
+									 replicationOriginSessionResetQuery->data);
+		connection->isReplicationOriginSessionSetup = false;
+	}
+}
+
+
+/*
+ * SetupReplicationOriginLocalSession sets up a new replication origin session in a
+ * local session.
+ */
+void
+SetupReplicationOriginLocalSession(void)
+{
+	/*elog(LOG, "Setting up local replication origin session"); */
+	if (!IsReplicationOriginSessionSetup(NULL))
+	{
+		repl_origin_local_session_start_no_publish();
+	}
+}
+
+
+/*
+ * ResetReplicationOriginLocalSession resets the replication origin session in a
+ * local node.
+ */
+void
+ResetReplicationOriginLocalSession(void)
+{
+	/*elog(LOG, "Resetting local replication origin session"); */
+
+	/*Reset Replication Origin in local session */
+	if (replorigin_session_origin != InvalidRepOriginId)
+	{
+		repl_origin_local_session_end_no_publish();
+	}
+}
+
+
+/*
+ * IsReplicationOriginSessionSetup checks if the replication origin is setup
+ * already in the local or remote session.
+ */
+static bool
+IsReplicationOriginSessionSetup(MultiConnection *connection)
+{
+	bool result = false;
+	if (connection == NULL)
+	{
+		result = repl_origin_local_session_is_no_publish();
+	}
+	else
```

Comment:
instead of having 2 different implementations in the same function, can we use 2 functions?

Reply:
Refactored this function into two different functions:
IsLocalReplicationOriginSessionActive and IsRemoteReplicationOriginSessionSetup